assets/src/components/shuttleMapPage.tsx

```diff
@@ -57,39 +74,160 @@ const ShuttleMapPage = (): ReactElement<HTMLDivElement> => {
   )
   const trainVehicles: TrainVehicle[] = allTrainVehicles(trainVehiclesByRouteId)
 
+  const mobileMenuClass = mobileMenuIsOpen ? "blurred-mobile" : ""
+
   const selectedShuttles: Vehicle[] = filterShuttles(
     shuttles || [],
     selectedShuttleRunIds
   )
 
-  const mobileMenuClass = mobileMenuIsOpen ? "blurred-mobile" : ""
-
-  const followerResetKey =
-    selectedShuttleRunIds === "all" ? "all" : selectedShuttleRunIds.join(",")
-
   return (
     <div className={`c-shuttle-map ${mobileMenuClass}`}>
       <ShuttlePicker shuttles={shuttles} />
+      <ShuttleMap
+        selectedShuttles={selectedShuttles}
+        selectedShuttleRunIds={selectedShuttleRunIds}
+        selectedVehicleId={selectedVehicleOrGhost?.id}
+        selectedRouteShapes={shapes}
+        trainVehicles={trainVehicles}
+        tileType={tileType}
+        setTileType={(tileType) =>
+          dispatch(setTileType("shuttleMap", tileType))
+        }
+        selectVehicle={(vehicle) => dispatch(selectVehicle(vehicle))}
+      />
+    </div>
+  )
+}
+
+interface ShuttleMapProps {
+  // TODO: revisit this prop
+  selectedShuttleRunIds: RunId[] | "all"
+  //
+  selectedVehicleId?: VehicleId
+  selectedShuttles: Vehicle[]
+  selectedRouteShapes?: Shape[]
+  trainVehicles?: TrainVehicle[]
+  tileType?: TileType
+  setTileType: (tileType: TileType) => void
+  selectVehicle: (vehicle: Vehicle | Ghost) => void
+}
+
+export const ShuttleMap = ({
+  selectedShuttles,
+  selectedShuttleRunIds,
+  selectedVehicleId,
+  selectedRouteShapes,
+  trainVehicles,
+  selectVehicle,
+  tileType,
+  setTileType,
+}: ShuttleMapProps) => {
+  const positions = selectedShuttles.map(vehicleToLeafletLatLng)
+
+  type FollowerController = false | "user-location" | "vehicle-location"
+  const [followerController, setFollowerController] =
+    useState<FollowerController>("vehicle-location")
+
+  const state = useInteractiveFollowerState(),
+    { setShouldFollow: setFollowActive } = state
 
-      <div className="c-shuttle-map__map">
-        <MapFollowingSelectionKey
-          selectedVehicleId={selectedVehicleOrGhost?.id}
-          vehicles={selectedShuttles}
-          shapes={shapes}
-          trainVehicles={trainVehicles}
-          onPrimaryVehicleSelect={(vehicle) => dispatch(selectVehicle(vehicle))}
-          selectionKey={followerResetKey}
-          tileType={tileType}
-        >
-          <LayersControl.WithTileContext
-            setTileType={(tileType: TileType) =>
-              dispatch(setTileType("shuttleMap", tileType))
-            }
-          />
-        </MapFollowingSelectionKey>
-      </div>
+  const setShouldFollow = (controller: FollowerController) => {
+    setFollowerController(controller)
+    setFollowActive(controller !== false ? true : false)
+  }
+
+  const [location, setLocation] = useState<null | GeolocationCoordinates>(null)
```

Comment:
**suggestion:** given that we're now dealing with both user and vehicle locations, I think it would be best to be explicit about which kind we're talking about when naming variables (like by calling this `userLocation`) for instance). Similarly, we use the term "positions" for vehicles and "location" for user in this component, and I'm sort of torn on whether it makes sense to standardize on one or the other or if there's a meaningful distinction.